js/app/modules/contentGroup/contentGroup.js

```diff
@@ -166,9 +166,14 @@ const ContentGroup = new Module.Class({
 
     _on_models_changed: function () {
         let models = this._selection.get_models();
-        if (models.length === 0 && this._no_results) {
+        this.visible = true;
+        if (models.length === 0) {
             this._arrangement.set_models([]);
-            this._stack.visible_child_name = NO_RESULTS_PAGE_NAME;
+            if (this._no_results) {
```

Comment:
Not sure I agree with this; shouldn't app writers take care that content groups have a no-results submodule if there is a chance that they can be empty?